sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/postgreSQL/PostgreCastToBoolean.scala

```diff
@@ -42,7 +42,7 @@ case class PostgreCastToBoolean(child: Expression, timeZoneId: Option[String])
   override def castToBoolean(from: DataType): Any => Any = from match {
     case StringType =>
       buildCast[UTF8String](_, str => {
-        val s = str.trim().toLowerCase()
+        val s = str.trimAll().toLowerCase()
```

Comment:
Are these really the only cases that should trim this way?

Reply:
the only left one is in string trim function, which I checked pg and hive are spaces trim only.

Reply:
presto string trim does java like way.

Reply:
Hm, it just seems odd to trim this just for booleans and dates?  is there more context here?

Reply:
for integral types, we currently handle relative issue here https://github.com/apache/spark/pull/26622